src/actions/bookmarks.js

```diff
@@ -0,0 +1,294 @@
+ *  @property {string} payload.message - Error message
+ *  @property {object} payload.error - Error object
+*/
+
+
+/**
+ *
+ *
+ * @export
+ * @param {string} jwt - access_token granted for the user
+ * @param {number} uesrID - id of user
+ * @param {BookmarkToBeCreated} bookmarkToBeCreated
+ * @returns
+ */
+export function createSingleBookmark(jwt, uesrID, bookmarkToBeCreated) {
+  /* go-api takes `published_date` as an unix timestamp (in secs) int */
+  const { published_date, ...passedBookmarkProperties } = bookmarkToBeCreated
+  return function (dispatch, getState, axiosInstanceWithUserCookie) {
+    const url = formAPIURL(apiEndpoints.createSingleBookmark(uesrID), false)
+    const axiosConfig = {
+      timeout: apiConfig.timeout,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${jwt}`
+      }
+    }
+    const interceptor = axiosInstanceWithUserCookie.interceptors.request.use((config) => {
+      const { method, url, headers, data } = config
+      dispatch({
+        type: types.singleBookmark.create.request,
+        payload: {
+          method,
+          url,
+          headers,
+          body: data
+        }
+      })
+      axiosInstanceWithUserCookie.interceptors.request.eject(interceptor)
+      return config
+    })
+    return axiosInstanceWithUserCookie.post(url, {
+      published_date: Math.ceil(new Date(published_date).getTime() / 1000),
+      ...passedBookmarkProperties
+    }, axiosConfig)
+      .then((res) => {
+        const successAction = buildSuccessActionFromRes(res, types.singleBookmark.create.success)
+        dispatch(successAction)
+      })
+      .catch((error) => {
+        const failAction = failActionCreators.axios(error, types.singleBookmark.create.failure)
+        dispatch(failAction)
+      })
+  }
+}
+
+
+/**
+ *
+ *
+ * @export
+ * @param {string} jwt - access_token granted for the user
+ * @param {number} userID - id of user
+ * @param {number} offset - the offset of the request
+ * @param {number} limit - max amount of records per fetch
+ * @param {string} sort - sort by
+ * @returns
+ */
+export function getMultipleBookmarks(jwt, userID, offset, limit, sort) {
+  return function (dispatch, getState, axiosInstanceWithUserCookie) {
+    const url = formAPIURL(`${apiEndpoints.getBookmarks(userID)}?offset=${offset}&limit=${limit}&sort=${sort}`, false)
+    const axiosConfig = {
+      timeout: apiConfig.timeout,
+      headers: {
+        Authorization: `Bearer ${jwt}`
+      }
+    }
+    const interceptor = axiosInstanceWithUserCookie.interceptors.request.use((config) => {
+      const { method, url, headers, data } = config
+      dispatch({
+        type: types.multipleBookMarks.read.request,
+        payload: {
+          method,
+          url,
+          headers,
+          body: data
+        }
+      })
+      axiosInstanceWithUserCookie.interceptors.request.eject(interceptor)
+      return config
+    })
```

Comment:
Interceptor is a good approach.
However, it is a one-time-used interceptor, which kind of loses its advantages.
Could we `dispatch(RequestAction)` directly?

Reply:
Because we may set the `headers` via the global axios generator, the custom axios instance and the methods of the instance, use the interceptor is the most certain way to catch the `headers` that axios actually use to send a request (rather than merge the configs above manually)